love2d: add Darwin platform support

DannyDannyDanny opened the pull request:
## Summary
Adds macOS/Darwin support to the Love2D package, enabling Love2D game development on nix-darwin systems.

## Changes
- Add conditional buildInputs for Linux vs Darwin platforms
- Use lua5_1 instead of luajit on Darwin per upstream recommendation
- Update platforms list to include Darwin
- Fix libtool bundle/dylib linking issues on Darwin
- Patch macOS function calls with stub implementations
- Add postFixup to correct dylib path references

Closes #369476

## Testing
- [x] Built successfully on aarch64-darwin
- [x] Tested basic functionality (`love --version` works)
- [x] Verified Love2D can run simple games
- [x] Ran `nixpkgs-review wip` (no rebuilds triggered)
- [x] Tested basic functionality of binary in `./result/bin/`

## Platform Testing
Built on platform:
- [ ] x86_64-linux
- [ ] aarch64-linux
- [ ] x86_64-darwin
- [x] aarch64-darwin

Note: Only have access to aarch64-darwin for testing. Would appreciate cross-platform testing help from maintainers with other Darwin systems.

## Things done
- [x] Fits CONTRIBUTING.md and pkgs/README.md guidelines
- [x] Ran `nixpkgs-review wip` on this change
- [x] Tested basic functionality of all binary files in `./result/bin/`

DannyDannyDanny committed:
Add Love2D Darwin contribution analysis

- Comprehensive analysis of making Love2D work on macOS/Darwin
- Includes technical approach, requirements, and testing steps
- TODO: Remove this file before final PR submission

DannyDannyDanny committed:
Update progress: branch setup complete

- Repository cloned and configured
- Branch renamed to love2d-darwin-support
- Ready to examine current Love2D package

UlyssesZh reviewed (commented):
Thank you. Could you please squash the commits?

Also I'm not sure whether not using the PR template fits the contribution guide, but it looks fine.

UlyssesZh commented:
You can ignore the failed builds of dependent packages on darwin. We should ask their respective maintainers to fix them. As for balatro, I think it may be because GitHub Actions were faulty because there is no build log. I will try building it on my own machine.

DannyDannyDanny commented on pkgs/development/interpreters/love/11.nix:
```diff
@@ -38,14 +39,14 @@ stdenv.mkDerivation rec {
-    libGL
     openal
-    luajit
+    (if stdenv.isDarwin then lua5_1 else luajit)
```
Yes. LuaJIT is the default upstream, but it’s not stable on macOS/aarch64: the JIT engine hits code-memory limits and has other long-standing issues on Apple Silicon. Using plain Lua 5.1 avoids those crashes/slowdowns while keeping full API compatibility with LÖVE (which still targets the 5.1 language). This matches upstream’s documented build option (“can be built against standard Lua instead of LuaJIT”) and also follows prior art in nixpkgs (e.g. Neovim prefers lua5_1 on Darwin).

Should I put a short comment in the code?

UlyssesZh replied:
Yeah just add a one-sentence comment saying that upstream says Lua 5.1 is preferred over LuaJIT, and also add a link to the upstream documentation if it is not easy to find.

DannyDannyDanny commented on pkgs/development/interpreters/love/11.nix:
```diff
@@ -56,22 +57,71 @@ stdenv.mkDerivation rec {
+      grep -n 'std::string("")' src/love.cpp || echo "Stub replacement verification failed"
+    else
+      echo "ERROR: macOS source file not found"
+    fi
```
done!

UlyssesZh commented:
Ah Balatro is a paid game, so let's ignore its failure in nixpkgs-review too.

UlyssesZh commented:
Yes, and force push.

DannyDannyDanny committed:
love2d: add Darwin platform support

- Add conditional buildInputs for Linux vs Darwin platforms
- Use lua5_1 instead of luajit on Darwin per upstream recommendation
- Update platforms list to include Darwin
- Fix libtool bundle/dylib linking issues on Darwin
- Patch macOS function calls with stub implementations
- Add postFixup to correct dylib path references

Fixes #369476

7c6f434c commented:
Oh, I meant force-pushing here would also be fine But no problem
